db/seeds.rb

```diff
@@ -97,503 +65,390 @@ def terms_of_service_text
+  res.due_date = time_rand(res.start_date.to_time, res.start_date.next_week, 
+                             checkout_length).to_datetime
+  res.checked_in = [nil, time_rand(random_time, random_time.next_week,
+                                     checkout_length).to_datetime].sample
+  res.checked_out = res.checked_in.nil? ? [nil, random_time.to_datetime].sample : random_time.to_datetime
+  res.equipment_object_id = res.checked_out ? res.equipment_object_id : nil
+  res.notes = Faker::HipsterIpsum.paragraph(8)
+  res.notes_unsent = [true, false].sample
+  begin
+    res.save!
+    if rand(0..3) == 0
+      # throw into the past
+      factor = rand(1.day..1.year)
+      res.start_date = res.start_date - factor
+      res.due_date = res.due_date - factor
+      res.save(validate: false)
+    end
+  rescue ActiveRecord::RecordInvalid
+    generate_reservation
+  end
 end
 
-unless entered_num == 0
-  if entered_num.integer? && entered_num > 0
-    progress = ProgressBar.create(format: progress_str, total: entered_num)
-    checkin_procedure = entered_num.times.map do
+
+def generate_objs(method, obj, n)
+  unless n == 0
+    progress = ProgressBar.create(format: PROGRESS_STR, total: n)
+    n.times.map do
       progress.increment
-      CheckinProcedure.create! do |chi|
-        chi.step = Faker::HipsterIpsum.sentence
-        chi.equipment_model_id = equipment_model.flatten.sample.id
-      end
+      send(method)
     end
-    puts "\n#{entered_num} check-in procedures successfully created!"
-  else
-    puts "\nPlease enter a whole number greater than 0."
-    entered_num = STDIN.gets.chomp.to_i
+    puts "\n#{n} #{obj} records successfully created!" 
   end
 end
 
-# CheckoutProcedure generation
-# ============================================================================
+# START SCRIPT
+# ============
+#
+
+MINIMAL = ENV["minimal"].present? || ENV["fast"].present?
+SEMI = ENV["semi"].present?
+NO_PICS = ENV["no_pics"].present? || ENV["fast"].present?
 
-if ENV["minimal"]
-  entered_num = 3
-else
-  entered_num = ask_for_records("CheckoutProcedure")
+if MINIMAL
+  puts "Minimal mode activated. Please wait..."
 end
 
-unless entered_num == 0
-  if entered_num.integer? && entered_num > 0
-    progress = ProgressBar.create(format: progress_str, total: entered_num)
-    checkout_procedure = entered_num.times.map do
-      progress.increment
-      CheckoutProcedure.create! do |cho|
-        cho.step = Faker::HipsterIpsum.sentence
-        cho.equipment_model_id = equipment_model.flatten.sample.id
-      end
+t1 = Time.now
+
+# check to see if a superuser exists
+if User.where('role = ?', 'superuser').empty?
+  User.destroy_all
+  u = User.new
+  u.first_name = "Donny"
+  u.last_name = "Darko"
+  u.phone = "6666666666"
+  u.email = "email@email.com"
+  u.affiliation = "Your Mother"
+  u.role = 'superuser'
+  u.view_mode = 'superuser'
+  u.username = 'dummy'
+  u.save
+
+  if MINIMAL || SEMI
+    if ENV['CAS_AUTH']
+      printf '(NetID)'
```

Comment:
Should we be renaming this to something more generic? IIRC 'NetID' is specifically a Yale term, maybe we should prompt them for their 'CAS Username' or something along those lines.